internal/test/ensure/ensure.go

```diff
@@ -36,9 +36,6 @@ func HelmHome(t *testing.T) func() {
 	os.Setenv(helmpath.CacheHomeEnvVar, "")
 	os.Setenv(helmpath.ConfigHomeEnvVar, "")
 	os.Setenv(helmpath.DataHomeEnvVar, "")
-	return func() {
-		os.RemoveAll(base)
```

Comment:
It might be useful to leave the directory in place if the test fails.

Reply:
indeed, but currently, the HelmHome() caller always call it as 'defer HelmHome(t)() ' which has the same behaviors with T.cleanup() i guess.

Reply:
Apologies; I wasn't clear with my comment.  I don't think that `helm` should keep the returned func (I like how it's cleaned up), but rather, in the func called by `t.Cleanup`, it might be good to check `t.Failed` before calling `os.RemoveAll`, because if a test fails, it could be useful to look in those directories for the state of config, cache, etc.  Maybe that's a different PR & discussion.

Reply:
thanks for your explanation, i just append a commit to do this